http/src/client/mod.rs

```diff
@@ -200,7 +200,18 @@ pub struct Client {
     pub(crate) application_id: AtomicU64,
     pub(crate) default_allowed_mentions: Option<AllowedMentions>,
     default_headers: Option<HeaderMap>,
-    http: HyperClient<HttpsConnector<HttpConnector>, Body>,
+    #[cfg(any(
+        feature = "native",
+        feature = "rustls-native-roots",
+        feature = "rustls-webpki-roots"
+    ))]
+    http: HyperClient<HttpsConnector<HttpConnector>>,
+    #[cfg(not(any(
+        feature = "native",
+        feature = "rustls-native-roots",
+        feature = "rustls-webpki-roots"
+    )))]
+    http: HyperClient<HttpConnector>,
```

Comment:
I'd prefer having the actual connector in a type alias, this is getting a bit big